cycle_en_terre_custom/models/stock_picking.py

```diff
@@ -0,0 +1,23 @@
+from odoo import api, fields, models
+
+
+class Picking(models.Model):
+    _inherit = "stock.picking"
+
+    origin_id = fields.Many2one(
+        string="Source Document",
+        comodel_name="sale.order",
+        compute="_compute_origin_id",
+    )
+    team_id = fields.Many2one(related="origin_id.team_id")
+
+    @api.multi
+    @api.depends("origin")
+    def _compute_origin_id(self):
+        self.ensure_one()
```

Comment:
FIXME:

```
  File "/home/odoo11/coopiteasy/cycle-en-terre/cycle_en_terre_custom/models/stock_picking.py", line 17, in _compute_origin_id
    self.ensure_one()
  File "/home/odoo11/odoo/odoo/models.py", line 4454, in ensure_one
    raise ValueError("Expected singleton: %s" % self)
ValueError: Expected singleton: stock.picking(11185, 11184)
```

Reply:
You can't expect to have a singleton in a _compute method. A little loop here would be enough:

```py
for picking in self:
  if picking.origin:
              picking.origin_id = self.env["sale.order"].search(
                  [("name", "like", picking.origin)], limit=1
              )
          else:
              picking.origin_id = False
```

or set it at invoice creation like here: https://github.com/coopiteasy/addons/pull/158/files